community-membership.md

```diff
@@ -0,0 +1,168 @@
+- **[Open an issue][membership request] against the sigstore/community repo**
+   - Ensure your sponsors are @mentioned on the issue
+   - Complete every item on the checklist ([preview the current version of the template][membership template])
+   - Make sure that the list of contributions included is representative of your work on the project.
+- Have your sponsoring reviewers reply confirmation of sponsorship: `+1`
+
+### Responsibilities and privileges
+
+- Responsive to issues and PRs assigned to them
+- Responsive to mentions of teams they are members of
+- Active owner of code they have contributed (unless ownership is explicitly transferred)
+  - Code is well tested
+  - Tests consistently pass
+  - Addresses bugs or issues discovered after code is accepted
+- They can be assigned to issues and PRs, and people can ask members for reviews with a `/cc @username`.
+- Tests can be run against their PRs automatically.
+
+**Note:** members who frequently contribute code are expected to proactively
+perform code reviews and work towards becoming a primary *reviewer* for the
+subproject that they are active in.
+
+## Reviewer
+
+Reviewers are able to review code for quality and correctness on some part of a
+subproject. They are knowledgeable about both the codebase and software
+engineering principles.
+
+**Defined by:** *reviewers* entry in an CODEOWNERS file in a repo owned by the
+Sigstore project.
+
+Reviewer status is scoped to a part of the codebase.
+
+**Note:** Acceptance of code contributions requires at least one approver in
+addition to the assigned reviewers.
+
+### Requirements
+
+The following apply to the part of codebase for which one would be a reviewer in
+an [CODEOWNERS] file (for repos using the bot).
+
+- member for at least 3 months
+- Primary reviewer for at least 5 PRs to the codebase
+- Reviewed or merged at least 20 substantial PRs to the codebase
+- Knowledgeable about the codebase
+- Sponsored by a project approver
+  - With no objections from other approvers
+  - Done through PR to update the OWNERS file
+- May either self-nominate, be nominated by an approver in this subproject, or be nominated by a robot
+
+### Responsibilities and privileges
+
+The following apply to the part of codebase for which one would be a reviewer in
+an [CODEOWNERS] file (for repos using the bot).
+
+- Tests are automatically run for PullRequests from members of the Sigstore GitHub organization
+- Code reviewer status may be a precondition to accepting large code contributions
+- Responsible for project quality control via [code reviews]
+  - Focus on code quality and correctness, including testing and factoring
+  - May also review for more holistic issues, but not a requirement
+- Expected to be responsive to review requests as per [community expectations]
+- Assigned PRs to review related to the project of expertise
+- Assigned test bugs related to the project of expertise
+- May get a badge on PR and issue comments
+
+## Approver
```

Comment:
Is there a reason we're splitting the two of these? All of the things you mention in approver, understanding API conventions, performance, interactions, etc, I think should be the responsibility of the reviewer role also.

There isn't a functional difference between the two in CODEOWNERS, right? I see a mention here that approvers are top-level, is the thought that reviewers would be per-directory?

My thought is these should be the same role in the community guidelines. You either can be a top-level codeowner if you have demonstrated an understanding of the entire project, or a per-directory/package codeowner if you have an understanding of a smaller part of the codebase.

Reply:
agree, with you

Reply:
the reviewer in our case can be more the triage role, will update that